### ISSUE TITLE
Should heading tags change font-size and other stylings?

### ISSUE THREAD

**Issue by loganmccaul** [carbon-design-system/carbon#99]
The heading tags (`<h1> - <h6>`) are used to establish a hierarchy of content on a page that can be utilized by screen readers to form page structure. https://www.w3.org/WAI/tutorials/page-structure/headings/

With the default stylings from [_css--body](https://github.com/carbon-design-system/carbon-components/blob/321b63e2d553b3f583af234b7c67b4f0a115c3d7/src/globals/scss/_css--body.scss) attached to the above tags currently a page's visuals can change drastically when restructuring the headings to be semantic. Furthermore, MDN recommends not using heading tags to change font size, https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements. 

Should the heading tags be changed so they just reset the elements default styles and then classes be created, `heading--1, heading--2, heading--*` to allow the appropriate stylings be attached to any element?

**Comment by chrisdhanaraj** [comment]
To clarify, the MDN tool recommends not using the heading tag to merely change font-size, but to make sure you're actually intending to lower the semantic value (their recommendation is to use the css attribute `font-size`). Prestyling the h1 - h6 tags with relevant font-sizes is intended and why the browser does it by default - h1's should be your largest, with h6 your smallest. 

There's an argument to also generating a `.heading-1` class to give visual styling to a semantic `h2` element (https://csswizardry.com/2012/02/pragmatic-practical-font-sizing-in-css), but in any scenario the default tags should be styled as well. 

in summary - yes the h1 - h6 should be prestyled, but we should also provide a class to let people quickly style h2's like an h1. Want to create an issue for that?

**Comment by loganmccaul** [comment]
I can do that.
